.github/workflows/build-images.yml

```diff
@@ -1,94 +1,59 @@
-          context: cc7_lcg95
-          buildargs: branch
-
-  cc7_lcg96:
-    runs-on: ubuntu-latest
-    needs: base_images
-    steps:
-      - uses: actions/checkout@v1
-      - name: Extract branch name
-        shell: bash
-        run: echo "::set-env name=branch::$(echo ${GITHUB_REF#refs/heads/} | sed 's/\//_/g')"
-      - uses: acts-project/Publish-Docker-Github-Action@master
-        with:
-          registry: gitlab-registry.cern.ch
-          name: acts/machines/cc7_lcg96
-          username: ${{ secrets.DOCKER_USERNAME  }}
-          password: ${{ secrets.DOCKER_PASSWORD  }}
-          context: cc7_lcg96
-          buildargs: branch
-
-  slc6_lcg95:
-    runs-on: ubuntu-latest
-    needs: base_images
-    steps:
-      - uses: actions/checkout@v1
-      - name: Extract branch name
-        shell: bash
-        run: echo "::set-env name=branch::$(echo ${GITHUB_REF#refs/heads/} | sed 's/\//_/g')"
-      - uses: acts-project/Publish-Docker-Github-Action@master
-        with:
-          registry: gitlab-registry.cern.ch
-          name: acts/machines/slc6_lcg95
-          username: ${{ secrets.DOCKER_USERNAME  }}
-          password: ${{ secrets.DOCKER_PASSWORD  }}
-          context: slc6_lcg95
-          buildargs: branch
-
-  cc7_clang_tidy:
-    runs-on: ubuntu-latest
-    needs: base_images
+    strategy:
+      matrix:
+        directory:
+          - centos7
+          - check-format
+          - fedora31
+          - slc6
     steps:
-      - uses: actions/checkout@v1
-      - name: Extract branch name
+      - name: 'Extract image tag'
         shell: bash
-        run: echo "::set-env name=branch::$(echo ${GITHUB_REF#refs/heads/} | sed 's/\//_/g')"
+        # use extraction command from Publish-Docker-Github-Action
+        run: echo "::set-env name=TAG::$(echo ${GITHUB_REF} | sed -e 's/refs\/heads\///g' | sed -e 's/\//-/g')"
+      - uses: actions/checkout@v2
       - uses: acts-project/Publish-Docker-Github-Action@master
         with:
-          registry: gitlab-registry.cern.ch
-          name: acts/machines/static_analysis/cc7-clang-tidy
-          username: ${{ secrets.DOCKER_USERNAME  }}
-          password: ${{ secrets.DOCKER_PASSWORD  }}
-          context: check_llvm8
-          buildargs: branch
-
-  base_images:
+          registry: docker.pkg.github.com
+          name: ${{ github.repository }}/${{ matrix.directory }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          context: ${{ matrix.directory }}
+          tags: ${{ env.TAG }}
+  derived-image:
     runs-on: ubuntu-latest
+    needs: base-image
     strategy:
       matrix:
-        dir:
-          - cc7
+        directory:
+          - centos7-lcg95apython3
+          - centos7-lcg96
+          - fedora31-hep
           - slc6
-          - ubuntu
-          - ubuntu_llvm9
-          - check
-          - check_llvm8
+    env:
+      REGISTRY: ${{ github.repository }}
```

Comment:
I understand what you want to do here (and I had missed this bit before), but that's not the registry.

Reply:
Fixed.